src/landscape.cpp

```diff
@@ -1285,6 +1292,158 @@ static void CreateRivers()
 		RunTileLoop();
 	}
 }
+#endif
+
+struct RiverData {
+	RiverData(): t(0), flowing(false), flow_dir(DIAGDIR_BEGIN), area(0), distance(0), flow_amount(0), visited(0) { prio = (Random() & 0xF); }
+	TileIndex t;
+	bool flowing;
+	DiagDirection flow_dir;
+	int area;
+	int distance;
+	int flow_amount;
+	uint visited;
+	uint prio;
+};
+void FloodFill(TileIndex tile, std::vector<RiverData> &riverdata, uint &count, uint mode) {
+	//can't use pointer to RiverData here everywhere, because rd->t is not fully initialized
+	RiverData *rd = &riverdata[tile];
+	if (rd->visited == mode) return;
+	rd->visited = mode;
+	if (!IsValidTile(tile)) {
```

Comment:
Is riverdata[INVALID_TILE] valid?

Reply:
it should only access the border of void-tiles